### PR TITLE
Update test runners and fix new compiler warning

### PR DESCRIPTION
**Pull request by mungre**
Update the tests to use the latest versions of Ubuntu and macOS.  Also fix the code to avoid new compiler warning.

### CODE REVIEW

File: src/value.h [hunk 1]
```diff
@@ -40,7 +40,6 @@ struct StringHeader
 		StringHeader* header = Allocate(length);
 		char* buffer = StringBuffer(header);
 		memcpy(buffer, text, length);
-		buffer[length] = 0;
```
Comment (chriskillpack) [inline src/value.h:43]: At first this made my Spidey Sense ™️ tingle until I saw the zero termination is done in `Allocate`.

File: src/value.h [hunk 2]
```diff
@@ -87,7 +86,6 @@ struct StringHeader
 			char* buffer = StringBuffer(header);
 			memcpy(buffer, StringData(header1), header1->m_length);
 			memcpy(buffer + header1->m_length, StringData(header2), header2->m_length);
-			buffer[length] = 0;
```
Comment (chriskillpack) [inline src/value.h:90]: Same as above. I _think_ this is okay because of the `header = Allocate(`.

**Review by chriskillpack — APPROVED** [review]
LGTM w/ comments

**Comment by mungre** [comment]
Cheers.  I tried [another way of fixing this](https://github.com/stardot/beebasm/tree/mungre/update-test-runners5) that I thought would be more readable but the compiler didn't like it.  So I think we're stuck with having the null termination out of sight in `Allocate`.

Do you want to put this in v1.10?

**Comment by chriskillpack** [comment]
I think it should go in - it's a small change that addresses scary warnings from clang. Go ahead and merge when you are ready.

**Pull request merged**